src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -50,6 +52,7 @@ public void Load(IEnumerable<string> paths)
             }
         }
 
+        [Obsolete("Unused and will be removed in a future version.")]
```

Comment:
this seems to be used in the dotnet/sdk: https://github.com/dotnet/sdk/pull/42458#issuecomment-2263071081

Reply:
2 options for you.
* Supress the warning
* Move to PopulateItemGroups

I'll revert the obsolete attribute for now.

Reply:
https://github.com/NuGet/NuGet.Client/pull/5946

Reply:
Thanks. I think moving to PopulateItemGroups is the right approach going forward.